### ISSUE TITLE
Warnings for multiple definitions could be helpful

### ISSUE THREAD

**Issue by InfiniteNesLives** [cc65/cc65#995]
I recently had an issue where ca65 wasn't using the ZP variable I thought I was telling it to.  Problem was due to me accidentally forgetting to make my function's variable definitions local.  So, I was mistakenly making multiple definitions on the global level, and didn't realize it.  It is typical for C compilers to throw warnings in this situation, and it can be helpful.  Just wanted to point this out in the event someone is interested in adding the warning.  Please don't take this as a feature request as personally I'm okay with things as-is.  But, it could help others from going down the "WTH" rabbit hole, like I did..

EXAMPLE:
```asm
.segment "ZEROPAGE"
reg0:   .res 1 
reg1:   .res 1 

.segment "CODE"
func_foo:
.define temp reg0   ; this isn't a problem when made local
rts

func_bar:
.define temp reg1    ; this isn't a problem when made local
rts

func_broke:
adc reg0  ; this will result in using reg1 without warnings of multiple definitions above.
```

**Comment by greg-king5** [comment]
A define-style macro was being expanded before it could be checked for prior existence.

Fixed in commit 5109c0b68f7a2cd07ece07b6812affeab07ae17a.

**Comment by InfiniteNesLives** [comment]
Thanks for your time!
